### PR TITLE
[Fix] Fix VOCDataset print wrong class names (#9878)

### PR DESCRIPTION
**Pull request by LoSealL**
## Motivation

To fix a print error when evaluating VOCDataset (#9878).

## Modification

Assign `ds_name` by `self.CLASSES` directly without the condition on `year` attribute.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by BIGWangYuDong** [comment]
This is not a good way to fix this potential error. voc07 is still necessary when calculating AP:

image

Hence, seems it should keep voc07.

**Comment by LoSealL** [comment]
> This is not a good way to fix this potential error. voc07 is still necessary when calculating AP:
> 
> image
> 
> Hence, seems it should keep voc07.

Move `mode` as an input argument to `eval_map` and make its default value to `'area'`.

**Comment by BIGWangYuDong** [comment]
> > This is not a good way to fix this potential error. voc07 is still necessary when calculating AP:
> > image
> > Hence, seems it should keep voc07.
> 
> Move `mode` as an input argument to `eval_map` and make its default value to `'area'`.

Yep, that's a way to solve this problem, but you still need to check whether the dataset is VOC2007

**Comment by LoSealL** [comment]
I don't think it's a good idea to check dataset name (or type) in evaluating function, otherwise there are massive information or even conflict information in that scope.

Besides, we've already told the function to use mode `11points` to calculate AUC, and we do use this mode in VOCDataset iff it's voc2007, so if we are using `VOCDataset` of year 2007, then in the evaluation it must be voc07 and use the `11points` mode.

File: mmdet/core/evaluation/mean_ap.py [hunk 1]
```diff
@@ -572,6 +573,9 @@ def eval_map(det_results,
             Default: False.
         use_group_of (bool): Whether to use group of when calculate TP and FP,
             which only used in OpenImages evaluation. Default: False.
+        mode (str): 'area' or '11points', 'area' means calculating the area
+            under precision-recall curve, '11points' means calculating the
+            average precision of recalls at [0, 0.1, ..., 1]. Default: 'area'.
```
Comment (BIGWangYuDong) [inline mmdet/core/evaluation/mean_ap.py:578]:
```suggestion
            average precision of recalls at [0, 0.1, ..., 1]. Defaults to 'area'.
```

**Comment by BIGWangYuDong** [comment]
I think the logic is OK now, a single comment need fix

File: mmdet/core/evaluation/mean_ap.py (hunk 1 above)
Comment (LoSealL) [reply mmdet/core/evaluation/mean_ap.py:578]: Change to `Defaults to` style? But at line 575 and others they use `Default:` style

Comment (BIGWangYuDong) [reply mmdet/core/evaluation/mean_ap.py:578]: Yes, you can change all `Default:` to `Defaults to` in this file.

Comment (BIGWangYuDong) [reply mmdet/core/evaluation/mean_ap.py:578]: We are going to unify using `defaults to` in mmdet

Comment (LoSealL) [reply mmdet/core/evaluation/mean_ap.py]: I've changed all `"Default:"` to `"Defaults to"` in this file.

**Comment by BIGWangYuDong** [comment]
@LoSealL LGTM now. Would you mind having a test on VOC2007, to check whether it is OK? 

Also, lint and ci are failed. 

image

please follow https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md to fix them

**Comment by LoSealL** [comment]
> @LoSealL LGTM now. Would you mind having a test on VOC2007, to check whether it is OK?
> 
> Also, lint is failed.
> 
> image
> 
> please follow https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md#2-configure-pre-commit to fix them

Fixing

**Review by BIGWangYuDong — APPROVED** [review]
LGTM

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**